### PR TITLE
Add registries warning to registries.conf

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Add a warning about fully specifying registries
to the search list.

Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@rhatdan PTAL

File: registries.conf [hunk 1]
```diff
@@ -8,6 +8,11 @@
 #
 # Registries to search for images that are not fully-qualified.
 # i.e. foobar.com/my_image:latest vs my_image:latest
+# NOTE: pulling an image that is not fully qualified, i.e. one that
```
Comment (rhatdan) [inline registries.conf]: I think this needs more description.  Let's use a google doc to start discussion.

**Comment by mtrmac** [comment]
I’m perfectly fine with adding a warning here, but note that the vast majority of users of short names will never read this config file, so warning _only_ here would not be sufficient.

**Comment by rhatdan** [comment]
Agreed, we also need to add this to container-registires.conf.md man page.

**Comment by TomSweeneyRedHat** [comment]
As the comments in the googledoc have died down, I've just updated this PR with the latest changes from there.  I did make some line feed changes for it to fit under 80 char, but left the wording intact.

**Comment by TomSweeneyRedHat** [comment]
And I think I made it past the validation test finally.

**Comment by rhatdan** [comment]
Can you add the same information into the man page for container-registries.conf.5.md

**Comment by vrothberg** [comment]
Once we consolidated the search-code into c/image, I'd love to throw a warning in the logs for that as well.

**Comment by TomSweeneyRedHat** [comment]
I've added to the man page too.  FWIW, I made one small change, at the start of the second paragraph we had "Note that it is recommended".  I changed that to "It is recommended".  We'd used the word in another place or two and it just sounded klunky to me.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**